ambry-tools/src/main/java/com.github.ambry.tools/perf/rest/NettyPerfClient.java

```diff
@@ -0,0 +1,501 @@
+              .ofType(String.class).defaultsTo("/random-id");
+      ArgumentAcceptingOptionSpec<Integer> concurrency =
+          parser.accepts("concurrency", "Number of parallel requests").withRequiredArg().describedAs("concurrency")
+              .ofType(Integer.class).defaultsTo(1);
+      ArgumentAcceptingOptionSpec<Long> postBlobTotalSize =
+          parser.accepts("postBlobTotalSize", "Total size in bytes of blob to be posted (-1 if GET is desired)")
+              .withRequiredArg().describedAs("postBlobTotalSize").ofType(Long.class).defaultsTo(-1L);
+      ArgumentAcceptingOptionSpec<Integer> postBlobChunkSize =
+          parser.accepts("postBlobChunkSize", "Size in bytes of each chunk that will be POSTed  (-1 if GET is desired)")
+              .withRequiredArg().describedAs("postBlobChunkSize").ofType(Integer.class).defaultsTo(-1);
+
+      OptionSet options = parser.parse(args);
+      this.host = options.valueOf(host);
+      logger.info("Host: {}", this.host);
+      this.port = options.valueOf(port);
+      logger.info("Port: {}", this.port);
+      this.path = options.valueOf(path);
+      logger.info("Path: {}", this.path);
+      this.concurrency = options.valueOf(concurrency);
+      logger.info("Concurrency: {}", this.concurrency);
+      this.postBlobTotalSize = options.valueOf(postBlobTotalSize);
+      logger.info("Post blob total size: {}", this.postBlobTotalSize);
+      this.postBlobChunkSize = options.valueOf(postBlobChunkSize);
+      logger.info("Post blob chunk size: {}", this.postBlobChunkSize);
+    }
+  }
+
+  /**
+   * Invokes the {@link NettyPerfClient} with the command line arguments.
+   * @param args command line arguments.
+   */
+  public static void main(String[] args) {
+    try {
+      ClientArgs clientArgs = new ClientArgs(args);
+      final NettyPerfClient nettyPerfClient =
+          new NettyPerfClient(clientArgs.host, clientArgs.port, clientArgs.path, clientArgs.concurrency,
+              clientArgs.postBlobTotalSize, clientArgs.postBlobChunkSize);
+      // attach shutdown handler to catch control-c
+      Runtime.getRuntime().addShutdownHook(new Thread() {
+        public void run() {
+          logger.info("Received shutdown signal. Requesting NettyPerfClient shutdown");
+          nettyPerfClient.shutdown();
+        }
+      });
+      nettyPerfClient.start();
+      nettyPerfClient.awaitShutdown();
+    } catch (Exception e) {
+      logger.error("Exception during execution of NettyPerfClient", e);
+    }
+  }
+
+  /**
+   * Creates an instance of NettyPerfClient
+   * @param host host to contact.
+   * @param port port to contact.
+   * @param path resource path.
+   * @param concurrency number of parallel requests.
+   * @param totalSize the total size in bytes of a blob to be POSTed (-1 if GET is intended).
+   * @param chunkSize size in bytes of each chunk to be POSTed (-1 if GET is intended).
```

Comment:
My thought was that it was an unnecessary argument but I see what you mean - it is much easier to understand if there is an explicit argument. I have added it now.